activemodel/lib/active_model/serialization.rb

```diff
@@ -145,6 +146,11 @@ def serializable_hash(options = nil)
         end
       end
 
+      rename_hash = options[:rename] || {}
+      rename_hash = rename_hash.stringify_keys
+      rename_hash = rename_hash.transform_values(&:to_s)
+      rename_hash.each { |key, value| hash[value] = hash.delete key }
```

Comment:
Ah yeah, looping through the renames does simplify things.

I think we can go a bit further: because we no longer need to lookup `rename_hash` by key, we can transform the keys as we encounter them, so no need to build the extra temporary hash:

```suggestion
      options[:rename]&.each do |old_name, new_name|
        old_name = old_name.to_s
        hash[new_name.to_s] = hash.delete(old_name) if hash.key?(old_name)
      end
```

Reply:
oh I was annoyed by that too, your way solves it elegantly. thanks a lot!